Sources/BraveWallet/Crypto/Stores/KeyringStore.swift

```diff
@@ -366,7 +366,7 @@ extension KeyringStore: BraveWalletKeyringServiceObserver {
     Task { @MainActor in
       let previouslySelectedCoin = await walletService.selectedCoin()
       walletService.setSelectedCoin(coinType)
-      if previouslySelectedCoin != coinType {
+      if previouslySelectedCoin != coinType || selectedAccount.coin != coinType {
```

Comment:
`selectedAccount.coin != coinType` will take care of the case that a new keyring has been created.